server/sonar-process/src/main/java/org/sonar/process/ProcessProperties.java

```diff
@@ -21,71 +21,120 @@
-  public static final String CLUSTER_NODE_TYPE = "sonar.cluster.node.type";
-  public static final String CLUSTER_SEARCH_HOSTS = "sonar.cluster.search.hosts";
-  public static final String CLUSTER_HOSTS = "sonar.cluster.hosts";
-  public static final String CLUSTER_NODE_PORT = "sonar.cluster.node.port";
-  public static final int CLUSTER_NODE_PORT_DEFAULT_VALUE = 9003;
-  public static final String CLUSTER_NODE_HOST = "sonar.cluster.node.host";
-  public static final String CLUSTER_NODE_NAME = "sonar.cluster.node.name";
-  public static final String CLUSTER_NAME = "sonar.cluster.name";
-  public static final String CLUSTER_WEB_STARTUP_LEADER = "sonar.cluster.web.startupLeader";
+  public enum Property {
+    JDBC_URL("sonar.jdbc.url"),
+    JDBC_USERNAME("sonar.jdbc.username", ""),
+    JDBC_PASSWORD("sonar.jdbc.password", ""),
+    JDBC_DRIVER_PATH("sonar.jdbc.driverPath"),
+    JDBC_MAX_ACTIVE("sonar.jdbc.maxActive", "60"),
+    JDBC_MAX_IDLE("sonar.jdbc.maxIdle", "5"),
+    JDBC_MIN_IDLE("sonar.jdbc.minIdle", "2"),
+    JDBC_MAX_WAIT("sonar.jdbc.maxWait", "5000"),
+    JDBC_MIN_EVICTABLE_IDLE_TIME_MILLIS("sonar.jdbc.minEvictableIdleTimeMillis", "600000"),
+    JDBC_TIME_BETWEEN_EVICTION_RUNS_MILLIS("sonar.jdbc.timeBetweenEvictionRunsMillis", "30000"),
+    JDBC_EMBEDDED_PORT("sonar.embeddedDatabase.port"),
+
+    PATH_DATA("sonar.path.data", "data"),
+    PATH_HOME("sonar.path.home"),
+    PATH_LOGS("sonar.path.logs", "logs"),
+    PATH_TEMP("sonar.path.temp", "temp"),
+    PATH_WEB("sonar.path.web", "web"),
+
+    SEARCH_HOST("sonar.search.host", InetAddress.getLoopbackAddress().getHostAddress()),
+    SEARCH_PORT("sonar.search.port", "9001"),
+    SEARCH_HTTP_PORT("sonar.search.httpPort"),
+    SEARCH_JAVA_OPTS("sonar.search.javaOpts", "-Xms512m -Xmx512m -XX:+HeapDumpOnOutOfMemoryError"),
+    SEARCH_JAVA_ADDITIONAL_OPTS("sonar.search.javaAdditionalOpts", ""),
+    SEARCH_REPLICAS("sonar.search.replicas"),
+    SEARCH_MINIMUM_MASTER_NODES("sonar.search.minimumMasterNodes"),
+    SEARCH_INITIAL_STATE_TIMEOUT("sonar.search.initialStateTimeout"),
+
+    WEB_JAVA_OPTS("sonar.web.javaOpts", "-Xmx512m -Xms128m -XX:+HeapDumpOnOutOfMemoryError"),
+    WEB_JAVA_ADDITIONAL_OPTS("sonar.web.javaAdditionalOpts", ""),
+    WEB_PORT("sonar.web.port"),
+
+    CE_JAVA_OPTS("sonar.ce.javaOpts", "-Xmx512m -Xms128m -XX:+HeapDumpOnOutOfMemoryError"),
+    CE_JAVA_ADDITIONAL_OPTS("sonar.ce.javaAdditionalOpts", ""),
+
+    HTTP_PROXY_HOST("http.proxyHost"),
+    HTTPS_PROXY_HOST("https.proxyHost"),
+    HTTP_PROXY_PORT("http.proxyPort"),
+    HTTPS_PROXY_PORT("https.proxyPort"),
+    HTTP_PROXY_USER("http.proxyUser"),
+    HTTP_PROXY_PASSWORD("http.proxyPassword"),
+    HTTP_NON_PROXY_HOSTS("http.nonProxyHosts"),
+    HTTP_AUTH_NLM_DOMAN("http.auth.ntlm.domain"),
+    SOCKS_PROXY_HOST("socksProxyHost"),
+    SOCKS_PROXY_PORT("socksProxyPort"),
+
+    CLUSTER_ENABLED("sonar.cluster.enabled", "false"),
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "false" 4 times. ([squid:S1192](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10300%2Fforbid_system_properties_in_settings_ws&issues=AWFNBHlqNbtQu1yJ8xlb&open=AWFNBHlqNbtQu1yJ8xlb)

[ISSUE_KEY]: AWFNBHlqNbtQu1yJ8xlb